Gems/Terrain/Code/Source/TerrainRenderer/Components/TerrainSurfaceMaterialsListComponent.cpp

```diff
@@ -112,6 +112,11 @@ namespace Terrain
     {
     }
 
+    TerrainSurfaceMaterialsListComponent ::~TerrainSurfaceMaterialsListComponent()
+    {
+        Deactivate();
+    }
```

Comment:
Why is this necessary?

It's to be expected that the components are deactivated by the component system. Is it safe to call the code from deactivation at destruction time? There are broadcast calls and buses disconnections happening during deactivation.
What happens if the component was already deactivated and then in the destructor it's deactivated again? Should it be protected to not deactivate it if it's already deactivated?

Is there anything in particular from the deactivation code that's supposed to be done in destruction too? If there is, then maybe that particular code could be put into a function so both the deactivation and destructor calls it.

Reply:
It's not now I've fixed the bug properly...